File: 05_hash_tables/go/02_check.go
/*
 * @Description: In User Settings Edit
 * @Author: your name
 * @Date: 2019-08-23 20:43:17
 * @LastEditTime: 2019-08-23 20:45:59
 * @LastEditors: Please set LastEditors
 */
package main

import "fmt"

var voted map[string]bool

func check_voter(name string) {
	if voted[name] {
		fmt.Println("存在")
	} else {
		voted[name] = true
		fmt.Println("不存在")
	}
}

func main() {
	voted = make(map[string]bool)
	check_voter("tom")
	check_voter("mike")
	check_voter("mike")
}

File: 05_hash_tables/ts/02_check.ts
/*
 * @Description: In User Settings Edit
 * @Author: your name
 * @Date: 2019-08-23 20:36:58
 * @LastEditTime: 2019-08-23 20:38:31
 * @LastEditors: Please set LastEditors
 */
let obj2 = {}

const check2 = (name:any):void =>  {
    if(obj2[name]){
        console.log('存在')
    }else{
        obj2[name] = true;
        console.log('不存在')
    }
}

check2('a')
check2('v')
check2('a')